packages/webview_flutter/webview_flutter/lib/src/types/webview_request.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+/// Defines the supported HTTP methods for loading a page in the [WebView].
+enum WebViewLoadMethod {
+  /// HTTP GET method.
+  get,
+
+  /// HTTP POST method.
+  post,
+}
+
+/// Extension methods on the [WebViewLoadMethod] enum.
+extension WebViewLoadMethodExtensions on WebViewLoadMethod {
+  /// Converts [WebViewLoadMethod] to [String] format.
+  String serialize() {
+    switch (this) {
+      case WebViewLoadMethod.get:
+        return 'get';
+      case WebViewLoadMethod.post:
+        return 'post';
+    }
+  }
+}
+
+/// Defines the parameters that can be used to load a page in the [WebView].
+class WebViewRequest {
+  /// Creates the [WebViewRequest].
+  WebViewRequest({
+    required this.method,
+    this.headers,
+    this.body,
+  });
+
+  /// HTTP method used to load the page.
+  final WebViewLoadMethod method;
+
+  /// HTTP headers for the request.
+  final Map<String, String>? headers;
```

Comment:
Generally we prefer default-empty collections rather than nullable collections, so there aren't two ways to express the same thing ("no headers" in this case).